lib/util/rfqValidator.ts

```diff
@@ -0,0 +1,104 @@
+import Logger from "bunyan";
+import { ethers, BigNumber } from "ethers";
+import { QuoteRequestData } from "../entities";
+import { RfqResponse } from "../handlers/quote";
+import { PermissionedTokenValidator } from "@uniswap/uniswapx-sdk";
+
+export class RFQValidator {
+
+  /**
+   * Validates if a token requires permission checks and if so, performs the preTransferCheck
+   * @param tokenAddress - The address of the token to validate
+   * @param chainId - The chain ID where the token exists
+   * @param from - The address tokens are being transferred from
+   * @param to - The address tokens are being transferred to
+   * @param amount - The amount of tokens being transferred (as a string)
+   * @param provider - Optional JsonRpcProvider needed for permissioned token checks
+   * @returns A string containing an error message if validation fails, undefined if successful
+   */
+  private static async validatePermissionedToken(
+    tokenAddress: string,
+    chainId: number,
+    from: string,
+    to: string,
+    amount: string,
+    provider?: ethers.providers.JsonRpcProvider
+  ): Promise<string | undefined> {
+    if (!PermissionedTokenValidator.isPermissionedToken(tokenAddress, chainId)) {
+      return undefined;
+    }
+
+    if (!provider) {
+      return `provider is required for permissioned token check for token: ${tokenAddress} on chain: ${chainId}`;
```

Comment:
if we end up doing custom errors would be nice to add them here to make the typing more strict

Reply:
LMK if this is what you had in mind